File: counting_triangles.cpp
int Solution::nTriang(vector<int> &A) {
    sort(A.begin(),A.end());
    int n = A.size(), res = 0, j, k;
    for(int i = n - 1;i>1;i--) {
        j = 0;
        k = i - 1;
        if(A[i] == 0)
            continue;
        while(j < k) {
            if(A[j] + A[k] > A[i]) {
                res += k - j;
                k--;
            } else
                j++;
            res %= 1000000007;
        }
    }
    return res;
}

File: counting_triangles.py
class Solution:
# @param A : list of integers
# @return an integer
    def nTriang(self, A):
        A.sort()
        res, n = [], len(A)
        for i in range(n - 2):
            j, k = i + 1, n - 1
            while k > j:
                if A[j] + A[k] > A[k]:
                    res.append(k - j)
                    k -= 1
                else:
                    j += 1
        return sum(res) % 1000000007